### DISCUSSION TITLE
input command missing? (to send keyevents)

### DISCUSSION THREAD

**Discussion by eadmaster** [termux/termux-app#3093] [Q&A] [answered]
Also asked [here](https://www.reddit.com/r/termux/comments/m3rjvv/call_input_to_simulate_input_events_in_termux/)

The `input` command seems missing, and non-root users cannot access `/system/bin/input`.

**Comment by agnostic-apollo** [comment, marked as answer]
Even if `input` command was provided in `$PATH` under `$PREFIX/bin`, like `logcat`, it would just be a wrapper around `/system/bin/input`. Moreover, `input` commands capabilities are very limited without being run as adb `shell` or `root` user, termux can't do anything about that.

There could be a `termux-api` provided that uses accessibility service like `AutoInput` app, but that would take lot of work and no plans for it currently.

**Comment by eadmaster** [comment]
I've just tried this in the [Terminal Emulator app](https://play.google.com/store/apps/details?id=jackpal.androidterm)  as non-root and it works:
````
$ input keyevent 85  # play_pause
````

I think a wrapper in `$PREFIX/bin` makes more sense.

**Reply by agnostic-apollo** [reply]
I said limited, you won't be able to inject events while app is not in foreground, etc.

That's `termux-tools` department.

https://github.com/termux/termux-tools/blob/e803f51a222bcedf7289fa15af49a1f49214a060/scripts/Makefile.am#L84

Can also just use `tudo input`.

https://github.com/agnostic-apollo/tudo
